// rfcontroljs/lib/protocols/weather12.js
module.exports = function(helper) {
  var protocolInfo, pulsesToBinaryMapping;
  pulsesToBinaryMapping = {
    '01': '0',
    '02': '1',
    '03': ''
  };
  return protocolInfo = {
    name: 'weather12',
    type: 'weather',
    values: {
      id: {
        type: "number"
      },
      channel: {
        type: "number"
      },
      temperature: {
        type: "number"
      }
    },
    brands: ["Nexus"],
    pulseLengths: [492, 969, 1948, 4004],
    pulseCount: 74,
    decodePulses: function(pulses) {
      var binary, channel, id, result, temperature;
      binary = helper.map(pulses, pulsesToBinaryMapping);
      id = helper.binaryToNumber(binary, 0, 9);
      channel = helper.binaryToNumber(binary, 10, 11) + 1;
      temperature = helper.binaryToSignedNumber(binary, 12, 23) / 10;
      humidity = helper.binaryToNumber(binary, 28, 35);
      return result = {
        id: id,
        channel: channel,
        temperature: temperature,
        humidity: humidity
      };
    }
  };
};

// 10001101 10000000 11001010 11110100 1010
// xxxxxxxx xx                                 ID          - 566
//            xx                               Channel     - 0 (+1 = 1)
//              xxxx xxxxxxxx                  Temperature - 202 (2002/10 = 20.2)
//                            xxxx             Unknown
//                                xxxx xxxx    Humidity    - 74
